### PR TITLE
chore: icons added for auth, security, traffic, serverless

### LINKED ISSUE #1760: support icons on Plugin Marcket
# Feature request

## Please describe your feature

Hi, we have some plugin icons for a few plugins currently, like this one:

image

To have a better & clearer view, we would better add plugins for those icons.

## Describe the solution you'd like

We could find some icons from `iconfont.cn`, but most can not, so we have to draw some icons by ourselves.

## Describe alternatives you've considered

None

## Additional context

We need icons in SVG format.

### PR DESCRIPTION
**Pull request by iamayushdas**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**
New icons added for 
- authentication
- security
- traffic
- serverless

Please update this section with detailed description.
Added the icons 

**Related issues**

fix/resolve #1760

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by juzhiyuan** [comment]
**Chinese**
image

**English**
image

**Comment by iamayushdas** [comment]
@tokers @liuxiran need your help in rectifying the CI issue

**Comment by juzhiyuan** [comment]
Could you see the Log here? https://github.com/apache/apisix-dashboard/pull/1805/checks?check_run_id=2399683974

Please have a try to fix it :)

**Comment by 1502shivam-singh** [comment]
> image

Nice 👍
Didn't knew some plugin-icons were already designed.

Something to mention (I should have mentioned this earlier now I feel) we should keep icons with same prefix or suffix together and have these ones with similar style like for `"serverless-post and pre functions"`. To explicitly mention here, by similar style I don't only mean color from the APISIX pallete or other, I mean similar color schemes and color softness, with focus on the idea that should resonate from that icon.

For example we have situation here that, all auth ones but jwt-auth have similar style.
- Easiest way to circumvent this for now is to redesign "jwt-auth" and make some small guide or better follow some already created one, to design rest of the icons (like the duotone style followed currently).

Also, something with sizes of the icons. Some look smaller than other like the `cors` and `wolf-rbac` one for instance.

**Comment by iamayushdas** [comment]
Thanks for your suggestion @1502shivam-singh smaller icons are the issue that would be resolved by using css properties, couldn't do the css stuff because of the urgency of icons last day,

Other icons are on the way as it takes a lot of time to think about every icon

**Comment by iamayushdas** [comment]
I am not sure if we are going to change the jwt-auth icon, let me complete all the icons first then will do the final touches, what say @1502shivam-singh

**Comment by 1502shivam-singh** [comment]
> Thanks for your suggestion @1502shivam-singh smaller icons are the issue that would be resolved by using css properties, couldn't do the css stuff because of the urgency of icons last day,
> 
> Other icons are on the way as it takes a lot of time to think about every icon

Yup , it's perfectly fine. You have got a lot of icons to design ;)


> I am not sure if we are going to change the jwt-auth icon, let me complete all the icons first then will do the final touches, what say @1502shivam-singh

Okay, we can do this later. I feel we should have the other icons on the dashboard too for a complete picture. Just keep in mind the "similar design" that I was implying in my above comment.

**Comment by guoqqqi** [comment]
Hi, @iamayushdas The failed E2E test in this PR I suggest fixing it like this.
image
This way, whenever you click on the Add button, you can be sure that the subsequent actions will be performed after you click on the Add button. Thanks~

**Comment by iamayushdas** [comment]
Okay

**Comment by iamayushdas** [comment]
@guoqqqi @juzhiyuan @spacewander @1502shivam-singh can i have review

File: web/cypress/integration/route/search-route.spec.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-loop-func */
+/* eslint-disable @typescript-eslint/no-invalid-this */
```
Comment (juzhiyuan) [inline web/cypress/integration/route/search-route.spec.js:2]: Hi, do we need this?

File: web/cypress/integration/route/search-route.spec.js [hunk 2]
```diff
@@ -53,10 +55,11 @@ context('Create and Search Route', () => {
 
       // eslint-disable-next-line no-loop-func
       cy.get(this.domSelector.drawerBody).within(() => {
-        cy.contains('Add').click();
-        cy.get(this.domSelector.labels_0_labelKey).type(`label${i}`);
+        cy.contains('Add').click().then(() => {
+          cy.get(this.domSelector.labels_0_labelKey).type(`label${i}`);
         cy.get(this.domSelector.labels_0_labelValue).type(`value${i}`);
```
Comment (juzhiyuan) [inline web/cypress/integration/route/search-route.spec.js:60]: It seems that there have some spaces?

File: web/src/components/Plugin/data.tsx [hunk 3]
```diff
@@ -25,6 +25,25 @@ export const PLUGIN_ICON_LIST: Record<string, any> = {
   'authz-keycloak': <IconFont name="iconkeycloak_icon_32px" />,
   'openid-connect': <IconFont name="iconicons8-openid" />,
   'kafka-logger': <IconFont name="iconApache_kafka" />,
+  'basic-auth': <IconFont name="iconbasic-auth" />,
```
Comment (juzhiyuan) [inline web/src/components/Plugin/data.tsx:28]: Good

File: web/cypress/integration/route/search-route.spec.js (hunk 1 above)
Comment (iamayushdas) [reply web/cypress/integration/route/search-route.spec.js:2]: yes otherwise it will end up with so much of warnings.

Comment (juzhiyuan) [reply web/cypress/integration/route/search-route.spec.js:2]: Do we have the same comments in other files? If no, we may need @guoqqqi to have a check.

Comment (iamayushdas) [reply web/cypress/integration/route/search-route.spec.js:2]: okay @guoqqqi could you please give a check to this

**Comment by iamayushdas** [comment]
All icons added now! 

scrnli_22_04_2021_19-58-03

File: web/cypress/integration/route/search-route.spec.js (hunk 1 above)
Comment (guoqqqi) [reply web/cypress/integration/route/search-route.spec.js]: I don't find the same comment elsewhere and I don't quite understand what this comment does, what type of warning would appear, please?

**Comment by guoqqqi** [comment]
LGTM! Thanks

**Comment by 1502shivam-singh** [comment]
> All icons added now!
> 
> scrnli_22_04_2021_19-58-03

Looks good. Just my previous concern of icons relative sizes (Maybe handle in a different PR or in this, as you wish) and the thing with "jwt-auth" icon if it could be redesigned.
Rest, LGTM.

**Comment by iamayushdas** [comment]
> > All icons added now!
> 
> > 
> 
> > scrnli_22_04_2021_19-58-03
> 
> 
> 
> Looks good. Just my previous concern of icons relative sizes (Maybe handle in a different PR or in this, as you wish) and the thing with "jwt-auth" icon if it could be redesigned.
> 
> Rest, LGTM. 

Will be adding your suggestion in next PR
THANK YOU

**Review by juzhiyuan — APPROVED** [review]
There are some details like color or icon meaning that need to be tweaked, but it's ok now. I recommend receiving more feedback & addressing these in the next PR.

**Comment by iamayushdas** [comment]
@LiteSun can u i have the review!

**Comment by nic-chen** [comment]
Is it too much colors..

**Comment by iamayushdas** [comment]
> Is it too much colors..

Your suggestions are welcome, but this is just the first update , just to have something in place of icon rather than nothing, after all the reviews i will be designing them all and include the collective suggestions from all.

**Review by nic-chen — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**